### PR TITLE
[Misc] Sort the list of embedding models

### PR DESCRIPTION
**Pull request by DarkLight1337**
This PR cleans up the list of embedding models by sorting them and removing the unnecessary merge function.

### CODE REVIEW

File: vllm/model_executor/models/registry.py [hunk 1]
```diff
@@ -94,33 +94,23 @@
...
     # [Text-only]
     "BertModel": ("bert", "BertEmbeddingModel"),
+    "DeciLMForCausalLM": ("decilm", "DeciLMForCausalLM"),
     "Gemma2Model": ("gemma2", "Gemma2EmbeddingModel"),
     "LlamaModel": ("llama", "LlamaEmbeddingModel"),
+    **{
+        # Multiple models share the same architecture, so we include them all
+        k: (mod, arch) for k, (mod, arch) in _TEXT_GENERATION_MODELS.items()
+        if arch == "LlamaForCausalLM"
+    },
```
Comment (russellb) [inline vllm/model_executor/models/registry.py:104]: The PR just mentions sorting the list, though this appears to add new elements to the list, as well. I assume that's intentional, but wanted to verify ...

Comment (DarkLight1337) [reply vllm/model_executor/models/registry.py:104]: These models were previously added via `add_embedding_models`. This dictionary comprehension replaces it.

Comment (russellb) [reply vllm/model_executor/models/registry.py:104]: Oh, I see, thanks!

**Review by russellb — APPROVED** [review]
The change lgtm. I also verified that the result is identical with the code before and after this change.

**Review by youkaichao — APPROVED** [review]
LGTM

**Pull request merged**